app/src/main/java/com/imashnake/animite/data/repos/MediaRepository.kt

```diff
@@ -11,7 +11,9 @@ class MediaRepository @Inject constructor(
 ) {
     suspend fun fetchMedia(id: Int?, mediaType: MediaType): MediaQuery.Media? {
         // TODO: Is there a better way to "wait for animations to complete"?
-        delay(100)
+        //  searchBarBottomPadding's animateDpAsState has a finishedListener, how do I propagate
```

Comment:
When you're faced with strange and complex logic like this, it usually means you need to rethink your approach to whatever you're doing in general
Of course, there are cases where complex logic is needed, but I don't think it's this :P
Why do we need to wait for animations to complete before navigating?